From b950d293e59a55753212ade1ffdcad0d828809e5 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Wed, 25 Jul 2018 15:55:21 +0530
Subject: [PATCH 0388/1852] drm: xlnx: vtc: Program Field 1 VTC registers in
 interlaced mode only

commit bd35510c9b36081c5c5fdc9313cf48f61629a17e from
https://github.com/Xilinx/linux-xlnx.git

The VTC has registers for field 0 and field 1. This patch allows
updating the field 1 registers only when in interlaced mode.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_vtc.c | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_vtc.c b/drivers/gpu/drm/xlnx/xlnx_vtc.c
index 97e739dffc28..228ce56fde39 100644
--- a/drivers/gpu/drm/xlnx/xlnx_vtc.c
+++ b/drivers/gpu/drm/xlnx/xlnx_vtc.c
@@ -264,10 +264,12 @@ static int xlnx_vtc_set_timing(struct xlnx_bridge *bridge,
 	xlnx_vtc_writel(vtc->base, XVTC_GVSHOFF_F0, reg);
 
 	/* Calculate and update Generator VBlank Hori field 1 */
-	reg = hactive & XVTC_XVXHOX_HSTART_MASK;
-	reg |= (hactive << XVTC_XVXHOX_HEND_SHIFT) &
-		XVTC_XVXHOX_HEND_MASK;
-	xlnx_vtc_writel(vtc->base, XVTC_GVBHOFF_F1, reg);
+	if (vm->flags & DISPLAY_FLAGS_INTERLACED) {
+		reg = hactive & XVTC_XVXHOX_HSTART_MASK;
+		reg |= (hactive << XVTC_XVXHOX_HEND_SHIFT) &
+			XVTC_XVXHOX_HEND_MASK;
+		xlnx_vtc_writel(vtc->base, XVTC_GVBHOFF_F1, reg);
+	}
 
 	/* Calculate and update Generator VBlank Hori field 1 */
 	if (vm->flags & DISPLAY_FLAGS_INTERLACED) {
@@ -279,7 +281,9 @@ static int xlnx_vtc_set_timing(struct xlnx_bridge *bridge,
 		reg |= (hsync_start << XVTC_XVXHOX_HEND_SHIFT) &
 			XVTC_XVXHOX_HEND_MASK;
 	}
-	xlnx_vtc_writel(vtc->base, XVTC_GVSHOFF_F1, reg);
+
+	if (vm->flags & DISPLAY_FLAGS_INTERLACED)
+		xlnx_vtc_writel(vtc->base, XVTC_GVSHOFF_F1, reg);
 
 	/* configure polarity of signals */
 	reg = 0;
-- 
2.31.1

